### PR TITLE
Add support for go modules/dep and github action go project tests

### LINKED ISSUE #2: Support go modules if a `go.mod` is present
Go modules is looking to be the standard tool for managing dependencies and will be built into the `go build` toolchain.

- https://github.com/golang/go/wiki/Modules

The action could support go modules if the action detects `go.mod`

If a repository has modules it is expected to run the build outside the $GOPATH, otherwise it has to inject environment variables to force the go tooling to use modules by default `GO111MODULES=on`.
> Invoke the go command in a directory outside of the $GOPATH/src tree, with a valid go.mod file in the current directory or any parent of it and the environment variable GO111MODULE unset (or explicitly set to auto).

Go team is planning on deprecating the `$GOPATH` in `go1.13` and making `GO111MODULES=on` by default.

Alternatively this could just be left to users defining their GO111MODULE in a Makefile or build script.

### PR DESCRIPTION
**Pull request by dougnukem**
Closes #2 

- Adds support for new Action environment variable `PROJECT_PATH` to allow for a subdirectory in the workspace to be used as Go project root (useful when multiple go builds are in a single mono repo, or for testing within this action)
- Adds test projects
  - `tests/projects/go_standard`
  - `tests/projects/go_modules`
  - `tests/projects/go_modules_vendor`
       - Adds a lot of files under `./vendor`

- Adds workflow actions to run those projects
```
action "Test Go Modules" {
  uses = "./"
  env = {
    PROJECT_PATH = "./tests/projects/go_modules"
    IMPORT = "cedrickring/golang-action"
  }
}
```

Using `act` local github action tool you can run the tests
```
$ act -a "Test Go Modules"
[Test Go Modules] docker build -t golang-action:3849038 /Users/ddaniels/dev/src/github.com/cedrickring/golang-action
[Test Go Modules] docker run image=golang-action:3849038 entrypoint=[] cmd=[]
go: finding github.com/spf13/viper v1.3.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/spf13/cobra v0.0.3
...
ok  	github.com/cedrickring/golang-action/tests/projects/go_modules	0.007s
?   	github.com/cedrickring/golang-action/tests/projects/go_modules/cmd	[no test files]

```

### CODE REVIEW

File: .github/main.workflow [hunk 1]
```diff
@@ -1,6 +1,12 @@
+<<<<<<< HEAD
```
Comment (cedrickring) [inline .github/main.workflow]: This file didnt get merged properly.

File: docker.mk [hunk 2]
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Comment (cedrickring) [inline docker.mk]: dito.

**Comment by dougnukem** [comment]
@cedrickring yeah adding support for `dep ensure` makes sense if we detect the presence of a `Gopkg.toml`

**Comment by dougnukem** [comment]
@cedrickring added support and tests for Go projects that use `dep` (e.g. presence of a `Gopkg.toml` and also if the dep dependencies are vendored ->`/vendor`)

File: go1.10/entrypoint.sh [hunk 3]
```diff
@@ -7,22 +7,49 @@ if [ -z "${IMPORT}" ]; then
...
+          # Run dep ensure to download and sync dependencies
+          "$GOPATH/bin/dep" ensure
+        fi
+      fi
+    fi
     go build ./...
     go test ./...
   fi
 else
+  echo "RUNNING STUFF: [$*]"
```
Comment (cedrickring) [inline go1.10/entrypoint.sh]: This line should be removed

File: go1.11/entrypoint.sh [hunk 4]
```diff
@@ -7,22 +7,49 @@ if [ -z "${IMPORT}" ]; then
...
+          # Run dep ensure to download and sync dependencies
+          "$GOPATH/bin/dep" ensure
+        fi
+      fi
+    fi
     go build ./...
     go test ./...
   fi
 else
+  echo "RUNNING STUFF: [$*]"
```
Comment (cedrickring) [inline go1.11/entrypoint.sh]: dito

File: go1.12/entrypoint.sh [hunk 5]
```diff
@@ -7,22 +7,49 @@ if [ -z "${IMPORT}" ]; then
...
+          # Run dep ensure to download and sync dependencies
+          "$GOPATH/bin/dep" ensure
+        fi
+      fi
+    fi
     go build ./...
     go test ./...
   fi
 else
+  echo "RUNNING STUFF: [$*]"
```
Comment (cedrickring) [inline go1.12/entrypoint.sh]: dito

**Comment by dougnukem** [comment]
@cedrickring cleaned up those debug echo statements

**Pull request merged**